engine/src/lib.rs

```diff
@@ -142,6 +144,20 @@ mod contract {
         io.return_output(state.owner_id.as_bytes());
     }
 
+    /// Set owner account id for this contract.
+    #[no_mangle]
+    pub extern "C" fn set_owner() {
+        let mut io = Runtime;
+        let mut state = state::get_state(&io).sdk_unwrap();
+        require_owner_only(&state, &io.predecessor_account_id());
+        let args = io.read_input().to_vec();
+        let sr: String = String::from_utf8(args).unwrap();
+        let id = AccountId::from_str(&sr).unwrap();
+        state.owner_id = id.clone();
+        state::set_state(&mut io, state).sdk_unwrap();
+        io.return_output(id.as_bytes());
```

Comment:
Rather than returning the input again, I think a simple `bool` would be fine (`true` for updating the value, `false` otherwise). This would avoid the clone when setting `state.owner_id`. In this scheme it also makes sense to check whether the new value differs from the old value so that `false` is actually returned sometimes.

Reply:
changing bool into bytes with `as_bytes` methods was not successful, so I made the method to return output as `&[1]`. I am not sure this will make sense though as `return_output` may act different than I expected. when I looked at the `SuccessValue` at workspace [test](https://github.com/hskang9/aurora-silo-tests/blob/11b5bb038d0c6c9422887ad546f355a8fa1033ea/src/test.rs#L156), it was returning nothing in SuccessValue.

```
Set Owner result: ExecutionFinalResult { total_gas_burnt: 5571972969272, transaction: ExecutionOutcome { block_hash: `ASrDKAigw1xeo4YmZCBuoP6M9zAgX11dNoEvR43VBS4j`, logs: [], receipt_ids: [`9ABq9UjKNjEgCAs1om4dHUsc86ivQMCvkzZ6U4JVEX51`], gas_burnt: 2427990313954, tokens_burnt: 242799031395400000000, executor_id: AccountId("owner.test.near"), status: SuccessReceiptId(9ABq9UjKNjEgCAs1om4dHUsc86ivQMCvkzZ6U4JVEX51) }, receipts: [ExecutionOutcome { block_hash: `ASrDKAigw1xeo4YmZCBuoP6M9zAgX11dNoEvR43VBS4j`, logs: [], receipt_ids: [`GRi9jNEWu9CU8kow4zKBLJJR1v1nRgcR5irhcr2gDa18`], gas_burnt: 2920800092818, tokens_burnt: 292080009281800000000, executor_id: AccountId("owner.test.near"), status: SuccessValue(``) }, ExecutionOutcome { block_hash: `GtjXTQbjfqdM7Lfhm2eaF8bz3tngBkNsf6kawW9pHHhJ`, logs: [], receipt_ids: [], gas_burnt: 223182562500, tokens_burnt: 0, executor_id: AccountId("owner.test.near"), status: SuccessValue(``) }], status: SuccessValue(``) }
```

Reply:
It would be helpful if I could find a way to display true in the return, should I put ascii bytes in "true" ?

Reply:
let data = true.try_to_vec().unwrap();
io.return_output(&data[..]);

Reply:
Ok thank you.

Reply:
Will indexing logic handle "true" as boolean value from result? I am not sure how this will go as I do not know the codec encode/decode of this.

Reply:
If indexing this tx result is unnecessary, I would just pass "true" as bytes. I need to confirm whether if this is ok.

Reply:
Added a comment about IO encoding: https://github.com/aurora-is-near/aurora-engine/pull/690#discussion_r1108502063